lib/view_component/engine.rb

```diff
@@ -22,7 +22,7 @@ class Engine < Rails::Engine # :nodoc:
         )
 
         if options.preview_path.present?
-          ActiveSupport::Deprecation.warn(
+          ViewComponent::Deprecation.warn(
             "`preview_path` will be removed in v3.0.0. Use `preview_paths` instead."
           )
```

Comment:
👋🏻 what do you think about inlining this check instead of making a new Deprecation class?

```suggestion
          ActiveSupport::Deprecation.warn(
            "`preview_path` will be removed in v3.0.0. Use `preview_paths` instead."
          ) unless ENV["VIEW_COMPONENT_ENV"] == "test"
```

Reply:
I'd say the only argument against it is the following:

We could look at asserting that deprecation warnings invoked. A separate class could-- instead of noop-ing-- retain the warning and allow us to write tests that it would warn outside of local test. So you could write a test that says something like, 
```ruby
# a bit pseudo codey
assert_deprecation_warning "`preview_path`...", preview_path`
```

I certainly get the inclination to avoid the indirection here though. So we could just go back to https://github.com/github/view_component/pull/586/commits/927c088cd3e5f7b8b5cfd500eca6c208e95e5438 if you prefer.

Reply:
The more I think about it, I think `ViewComponent::Deprecation` is actually a good case for a singleton class too 🤔

Reply:
All that said, if you prefer in-lining it then I'm 👍  on that too